### PR TITLE
gotosocial: 0.14.1 -> 0.14.2

### PR DESCRIPTION
**Pull request by blakesmith**
Another minor bugfix release: https://github.com/superseriousbusiness/gotosocial/releases/tag/v0.14.2.

I also added a simple test assertion that ensures we're getting back the expected running version of gotosocial in the NixOS test.

@misuzu Would you be open to having me on the maintainers list for this package? Would love to continue helping.

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by misuzu** [comment]
> Would you be open to having me on the maintainers list for this package? Would love to continue helping.

Please go ahead, thanks!

File: nixos/tests/web-apps/gotosocial.nix [hunk 1]
```diff
@@ -20,6 +20,9 @@
     machine.wait_for_unit("postgresql.service")
     machine.wait_for_open_port(8081)
 
+    # Ensure we're running the expected version of gotosocial
+    machine.succeed("curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.software.version' | grep -q '^\"${pkgs.gotosocial.version}\"'")
```
Comment (misuzu) [inline nixos/tests/web-apps/gotosocial.nix:24]: This should go into a separate commit (`nixos/gotosocial: ...`)

**Comment by blakesmith** [comment]
@misuzu Thanks for the feedback. I broke it up into two separate commits. Will open another separate PR to update maintainers.

**Comment by misuzu** [comment]
> Will open another separate PR to update maintainers.

You can do it right here, no need for a separate PR

**Comment by blakesmith** [comment]
@misuzu Updated the maintainers! Thanks for the help!

**Review by misuzu — APPROVED** [review]
LGTM! Thank you!

File: nixos/tests/web-apps/gotosocial.nix [hunk 2]
```diff
@@ -1,7 +1,7 @@
-{ lib, ... }:
+{ pkgs, lib, ... }:
```
Comment (drupol) [inline nixos/tests/web-apps/gotosocial.nix:1]:
```suggestion
{ lib, gotosocial, ... }:
```

**Review by drupol — CHANGES_REQUESTED** [review]
This PR should contain 3 commits only:

1. First one: `maintainers: add blakesmith`
2. The changes in `pkgs/servers/gotosocial/default.nix`
3. The changes in `nixos/*`

File: nixos/tests/web-apps/gotosocial.nix (hunk 2 above)
Comment (blakesmith) [reply nixos/tests/web-apps/gotosocial.nix:1]: @drupol I think I'm going to roll this test back. `gotosocial` isn't bound / passed into this function. I'm not sure this test has high value anyways: it feels like a bit of a smell to pass in `pkgs` to this test function.

```
error:
       … while evaluating the attribute 'drvPath'

         at /nix/store/k427qnlyr0ibh3qdc16zncxz1lxry5z7-source/lib/customisation.nix:268:7:

          267|     in commonAttrs // {
          268|       drvPath = assert condition; drv.drvPath;
             |       ^
          269|       outPath = assert condition; drv.outPath;

       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: attribute 'gotosocial' missing

       at /nix/store/k427qnlyr0ibh3qdc16zncxz1lxry5z7-source/lib/modules.nix:508:28:

          507|         builtins.addErrorContext (context name)
          508|           (args.${name} or config._module.args.${name})
             |                            ^
          509|       ) (lib.functionArgs f);
```

**Comment by blakesmith** [comment]
@drupol Thanks again for the feedback! I backed out the nixos test change for now, and cleaned up the commit organization per your suggestion. :-)

**Review by drupol — APPROVED** [review]

**Comment by drupol** [comment]
Argh I just missed it.

The commit log message AND the PR title should have been: `gotosocial: 0.14.1 -> 0.14.2`.
